slugid.js

```diff
@@ -46,6 +46,10 @@ exports.decode = function(slug) {
 exports.v4 = function() {
   var bytes   = uuid.v4(null, new Buffer(16));
   var base64  = bytes.toString('base64');
+  while (base64.substring(0, 1) == "+") {
+    bytes   = uuid.v4(null, new Buffer(16));
+    base64  = bytes.toString('base64');
+  }
```

Comment:
I think in js I also might be able to do something like this to avoid the repetition:

```
var bytes, base64;
do {
    bytes = ...
    base64 = ...
} while (base64.substring(0, 1) == "+");
```

... although this js newbie needs to check! =)